Order data validation and correction

akallabeth opened the pull request:
Related to fixing #4925 

* Adds checks to only allow orders that have been activated by client and server during capability exchange
* Various crash fixes found during fuzzing run
* Coordinate adjustments for orders, cut invalid ones before passing them on
* Split order handling in `support check`, `decoding` and `processing`

openconcerto commented:
This patch fixes memory corruption, but corrupts the display...

akallabeth commented:
@freerdp-bot test, come on!

MartinHaimberger commented on libfreerdp/core/orders.c:
```diff
@@ -3020,320 +3629,404 @@ static BOOL update_recv_primary_order(rdpUpdate* update, wStream* s, BYTE flags)
 	}
 
-	return TRUE;
+	return rc;
```
cant be false ... but ok

akallabeth commented:
@MartinHaimberger thx, removed the log messages and simplified the check.
Reading but not processing an order does not really make sense, so removed that option.

MartinHaimberger commented:
@freerdp-bot test

openconcerto commented:
Remmina is no more compatible with xrdp
BitBlt is broken.
Good job.
